site/profile/manifests/it/icinga_master.pp

```diff
@@ -0,0 +1,466 @@
+    git_revision   => 'v0.3.0',
+    require        => Class['::icingaweb2'],
+  }
+##IcingaWeb Incubator
+  class {'icingaweb2::module::incubator':
+    ensure         => present,
+    git_repository => 'https://github.com/Icinga/icingaweb2-module-incubator',
+    git_revision   => 'v0.5.0',
+    require        => Class['::icingaweb2'],
+  }
+##Icinga Director DB migration
+  exec { 'Icinga Director DB migration':
+    path    => '/usr/local/bin:/usr/bin:/bin',
+    command => 'icingacli director migration run',
+    onlyif  => 'icingacli director migration pending',
+    require => Class['icingaweb2::module::director'],
+  }
+  exec { 'Icinga Director Kickstart':
+    path    => '/usr/local/bin:/usr/bin:/bin',
+    command => 'icingacli director kickstart run',
+    onlyif  => 'icingacli director kickstart required',
+    require => Exec['Icinga Director DB migration'],
+  }
+
+##Nginx Resource Definition
+  nginx::resource::server { 'icingaweb2':
+    server_name          => [$ssl_fqdn],
+    ssl                  => true,
+    ssl_cert             => $ssl_cert,
+    ssl_key              => $ssl_key,
+    ssl_redirect         => true,
+    index_files          => ['index.php'],
+    use_default_location => false,
+    www_root             => '/usr/share/icingaweb2/public',
+    include_files        => ['/etc/nginx/sites-available/pnp4nagios.conf'],
+  }
+  nginx::resource::location { 'root':
+    location    => '/',
+    server      => 'icingaweb2',
+    try_files   => ['$1', '$uri', '$uri/', '/index.php$is_args$args'],
+    index_files => [],
+    ssl         => true,
+    ssl_only    => true,
+  }
+  nginx::resource::location { 'icingaweb':
+    location       => '~ ^/icingaweb2(.+)?',
+    location_alias => '/usr/share/icingaweb2/public',
+    try_files      => ['$1', '$uri', '$uri/', '/icingaweb2/index.php$is_args$args'],
+    index_files    => ['index.php'],
+    server         => 'icingaweb2',
+    ssl            => true,
+    ssl_only       => true,
+  }
+  nginx::resource::location { 'icingaweb2_index':
+    location      => '~ ^/index\.php(.*)$',
+    server        => 'icingaweb2',
+    ssl           => true,
+    ssl_only      => true,
+    index_files   => [],
+    try_files     => ['$uri =404'],
+    fastcgi       => '127.0.0.1:9000',
+    fastcgi_index => 'index.php',
+    fastcgi_param => {
+      'ICINGAWEB_CONFIGDIR' => '/etc/icingaweb2',
+      'REMOTE_USER'         => '$remote_user',
+      'SCRIPT_FILENAME'     => '/usr/share/icingaweb2/public/index.php',
+    },
+  }
+##Reload service in case any modification has occured
+# Run and Enable Service
+  service { 'php73-php-fpm':
+    ensure  => running,
+    require => Package[$packages],
+  }
+  service { 'npcd':
+    ensure  => running,
+    require => Package[$packages],
+  }
+  service { 'icinga-director':
+    ensure  => 'running',
+    require => Exec[$command3],
+  }
+#Force Deploy every puppet run
+  exec { $deploy_cmd:
```

Comment:
Note that it is better practice to create a cron job for tasks that need to be run periodically rather than relying on (and slowing down) the puppet agent.

Reply:
I'll see if I can created as part of a daemon, but I would really want to avoid cronjobs

Reply:
It's something that must be done or by hand or a curl. But if you think about it, if foreman (puppet-server) is down, all servers and VMs would be making much more noice than if icinga-master dies. Besides, having a "clicker" every 30 min wouldn't produce that much of an impact, specially in a single VM that coexists with rancher.